client/landing/gutenboarding/components/domain-picker-modal/index.tsx

```diff
@@ -3,47 +3,61 @@
  */
 import * as React from 'react';
 import Modal from 'react-modal';
+import { select } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import DomainPicker, { Props as DomainPickerProps } from '@automattic/domain-picker';
+import { recordEnterModal, recordCloseModal } from '../../lib/analytics';
+import { STORE_KEY } from '../../stores/onboard';
 
 /**
  * Style dependencies
  */
 import './style.scss';
 
-interface Props extends Omit< DomainPickerProps, 'tracksName' > {
+interface Props extends DomainPickerProps {
 	isOpen: boolean;
 	onMoreOptions?: () => void;
 }
 
-const DomainPickerModal: React.FunctionComponent< Props > = ( { isOpen, ...props } ) => {
+const DomainPickerModal: React.FunctionComponent< Props > = ( { isOpen, onClose, ...props } ) => {
 	// This is needed otherwise it throws a warning.
 	Modal.setAppElement( '#wpcom' );
 
-	if ( ! isOpen ) return null;
+	const tracksName = 'DomainPickerModal';
 
-	const handleAfterOpen = () => {
+	const handleOpen = () => {
 		// This fixes modal being shown without
 		// header margin when the modal is opened.
 		window.scrollTo( 0, 0 );
+		recordEnterModal( tracksName );
+	};
+
+	const handleClose = () => {
+		recordCloseModal( tracksName, {
+			selected_domain: select( STORE_KEY ).getSelectedDomain()?.domain_name,
+		} );
+		onClose && onClose();
 	};
 
+	if ( ! isOpen ) return null;
+
```

Comment:
An early return like this looks like it could go right at the top of the function. Is there any reason we need to do any work?

I'm surprised this doesn't require the block. I would've expected (and prefer):

```js
	if ( ! isOpen ) {
		return null;
	}
```

Coding guidelines do state that: https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md#blocks-and-curly-braces

Reply:
> An early return like this looks like it could go right at the top of the function. Is there any reason we need to do any work?

Ah yes, initially it was because of `useSelect` I had to move this down. Since `useSelect` is not necesary, this can be moved back up!

> Coding guidelines do state that: https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md#blocks-and-curly-braces

Okay! Gonna use the curly brackets! (Yeah, you'd expect maybe prettier would've prettified it but seems like it didn't).